### DISCUSSION TITLE
Role of HAProxy and blocked.ips

### DISCUSSION THREAD

**Discussion by GeoddHQ** [aptos-labs/aptos-developer-discussions#497] [Questions] [answered]
### Discord user ID

malith7574

### Describe your question in detail.

I’ve been looking through the node setup configurations, particularly around HAProxy and the use of the blocked.ips file for managing traffic. I noticed that it’s being used within the Dockerized environment, but I was curious about the larger goal behind this implementation.  Is the use of blocked.ips and HAProxy mainly intended to handle external traffic filtering, or is there a broader plan to integrate more comprehensive internal firewall rules within the node architecture?

### Which operating system are you using?

Linux (Ubuntu, Fedora, Windows WSL, etc.)

### Which SDK or tool are you using? (if any)

N/A

**Comment by Alexzander-Stone** [comment, marked as answer]
Hiya @GeoddHQ,

The main intention is to handle external traffic filtering, like you said. 
As of today, there aren't plans to expand it to what you're describing.

Do you have opinions or concerns on the direction we're taking?

**Reply by GeoddHQ** [reply]
Thanks for the clarification! It’s great to see that the current implementation using HAProxy is already addressing traffic limiting, filtering, and IP blocking, which is important. HAProxy’s lightweight nature makes it a solid choice for handling these tasks efficiently.

However, as Aptos continues to grow rapidly, especially as an L1 blockchain, I believe the security front for validators could benefit from more tailored solutions that integrate firewall rules, threat intelligence, and protocol-specific filtering directly into the node setup. While many datacenters provide some level of IDS/IPS for baremetal servers, relying solely on these external security measures may not be enough to ensure the robustness needed for a decentralized network like Aptos.

We’ll look to propose and test an AIP that enhances the current approach with more comprehensive security mechanisms, for the long-term safety and scalability of validator nodes. The idea would be to supplement the HAProxy setup with a deeper network-level defense, so as Aptos continues to scale, the integrity of L1 nodes can be independently protected, regardless of the datacenter’s infrastructure.

I think HAProxy is great for now, but Aptos’ rapid growth calls for proactive security steps to ensure it stays resilient.

Looking forward to your thoughts and on the approach to submit this feature once ready ?

**Reply by Alexzander-Stone** [reply]
Thanks for the great involvement on your parts, @GeoddHQ!

Always great to consider all factors to keep the network as resilient as ever. We'll be looking forward to seeing your AIP. 🤝
